variant_remapping_tools/replace_refs.py

```diff
@@ -3,25 +3,35 @@
 from argparse import RawTextHelpFormatter
 import linecache
 
-description = "replaces the REF allele with the old REF allele if REF = ALT in the remapped VCF"
 
-parser = argparse.ArgumentParser(description = description, formatter_class = RawTextHelpFormatter)
-parser.add_argument("-i", "--vcf", help = "VCF file containing remapped variants")
-parser.add_argument("-r", "--oldrefalleles", help = "file containing the same header as the VCF file + the old " 
-"REF alleles in the same order as the VCF")
-parser.add_argument("-o", "--outfile", help = "name of new file")
-args = parser.parse_args()
+def process_variant(line, line_num, oldrefalleles_file):
+    variant = line.split('\t')
+    # Gets corresponding REF allele from the oldrefalleles file.
+    old_ref = linecache.getline(oldrefalleles_file, line_num + 1).rstrip()
+    # The reference allele extracted from the new genome in insertReferenceAllele and stored in variant[3]
+    # is only one base long. There test "variant[3] == variant[4]" can only work if the ref and alt have
+    # the same length which won't be the case for indels.
+    # For insertion, we're leaving the single base extracted from the new genome (regardless of the alt)
+    # for deletion, we're using the old reference sequence (regardless of the alt)
+    # TODO: Correct this behaviour to have a consistent treatment of all variants types
+    if variant[3] == variant[4] or len(old_ref) != 1:
```

Comment:
I was wondering what happens with MNVs which could satisfy the old ref length criteria of not being equal to 1: 
![image](https://user-images.githubusercontent.com/25152678/108895342-5c49f280-760b-11eb-8bf4-3d2ed1cc2a58.png)

Reply:
I think MNV is a valid concern, but since, as you mentioned, they're quite rare, I think it's OK to address them in subsequent iterations. If we have/will have a backlog issues for this (I raised this here: https://github.com/EBIvariation/variant-remapping/pull/14#discussion_r581785078), we could document the need for processing MNVs in there as well.

Reply:
I created EVA-2328 to track the correction of the allele swap mechanisms.